### PR TITLE
Planning: Intro & Editors

### PR DESCRIPTION
**Pull request by philsturgeon**
Trying to teach  folks that editors are a thing so they don't moan about writing loads of YAML/JSON by hand. Who wants to do that?! >.<

This is the first chapter of "Part Two: Planning & Design" section, but there is a lot more content to go into Part One: Theory. That will cover the theory behind how to create good resources, what collections look like, how to name endpoints, UUIDs, etc. This is getting into the *How* rather heavily now.

Speaking of which, [preview the new chapters here](https://htmlpreview.github.io/?https://github.com/apisyouwonthate/build-apis-2/blob/planning/editors/generated/book.html#_part_two_planning_design).

### CODE REVIEW

**Comment by philsturgeon** [comment]
@bcrowe hey maybe you wanna take a look at this? Did you ever find an editor you liked?

File: 2-planning/0-introduction.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+When I was younger and dumber, the planning stage for APIs was pretty much
+"think of a list of endpoints then start coding". This is wildly insufficient.
+As the <<theory-contracts,API Contracts>> chapter explained, contracts are
+incredibly important throughout the life-cycle of an API. We are going to look
+into how they can be used in the planning and design stage.
+
+Most of the planning and design phase is intended to do two things:
+
+1. Make developers think really hard about the requirements for their API 2.
+Gather feedback from clients application developers 3. Use the feedback to
```
Comment (akrabat) [inline 2-planning/0-introduction.adoc]: `2.` and `3.` should be on separate lines…

**Comment by akrabat** [comment]
I think you need more between the intro section and the editors sections. This part is called _Planning & Design_ and we've jumped into how to do before we've learnt how to design.

It feels like this part is building on _1.2. API Contracts_ and I was hoping to learn how to go about the planning and design process. The editors section feels like it comes after that when we get to prototype our plan and create the contract.

File: 2-planning/0-introduction.adoc (hunk 1 above)
Comment (katzien) [inline 2-planning/0-introduction.adoc]:
```suggestion
Gather feedback from client application developers 3. Use the feedback to
```

File: 2-planning/0-introduction.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+improve the planning before the development phase begins
+
+== Time is Money
+
+Poorly planned APIs waste a lot of time. Developers cost money, so paying extra
+money for developers to rewriting a bunch of code they wrote just recently due
```
Comment (katzien) [inline 2-planning/0-introduction.adoc]:
```suggestion
for developers to rewrite a bunch of code they wrote just recently due
```

File: 2-planning/0-introduction.adoc (hunk 1 above)
Comment (philsturgeon) [reply 2-planning/0-introduction.adoc]: Ha thats my auto-wrap extension eatingn lines, thank you.

**Comment by philsturgeon** [comment]
@akrabat hey, totally understand why you think that. Check out the README for the overview, im not writing everything in order.

File: 2-planning/0-introduction.adoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+found to satisfy the needs of both clients, and the v1 could have been used for
+both. Now we've got two separate versions of the API being used at the same
+time, with developers maintaining two code paths, and at some point the original
+client using the original version of the API is going to have to rewrite their
+code to use the newer version of the API. What a daft mess.
+
+== Gather Feedback Early
+
+If clients had more insight into the planning of the API, they have a much
+better chance of catching out issues earlier on. We can give them fake API
```
Comment (chchrist) [inline 2-planning/0-introduction.adoc]: Who is "We"? Who writes the API contract, who creates the mock? Is the developer responsible to write the contract? Should a company have API specialists?

**Comment by bcrowe** [comment]
> @bcrowe hey maybe you wanna take a look at this? Did you ever find an editor you liked?

I've just been using Atom, handcrafting things, and using GitHub PRs for collaboration & discussions... which you have covered. Haven't given any of the OpenAPI-focused tooling a swing yet to be honest. Sorry!

File: 2-planning/0-introduction.adoc (hunk 3 above)
Comment (philsturgeon) [reply 2-planning/0-introduction.adoc]: @chchrist hey, I've added a commit which answers your question. Throughout the book im saying "we" as conversational, literally you and me.

Comment (chchrist) [reply 2-planning/0-introduction.adoc]: ah, cool 👍

**Pull request merged**